WebCryptoAPI/digest/digest.js

```diff
@@ -0,0 +1,162 @@
+
+            promise_test(function(test) {
+                var promise = subtle.digest({name: mixedCase}, sourceData[size])
+                .then(function(result) {
+                    assert_true(equalBuffers(result, digestedData[alg][size]), "digest() yielded expected result for " + alg + ":" + size);
+                }, function(err) {
+                    assert_unreached("digest() threw an error for " + alg + ":" + size + " - " + err.message);mixedCase
+                });
+
+                all_promises.push(promise);
+                return promise;
+            }, downCase + " with " + size + " source data");
+
+            promise_test(function(test) {
+                var promise = subtle.digest({name: mixedCase}, sourceData[size])
+                .then(function(result) {
+                    assert_true(equalBuffers(result, digestedData[alg][size]), "digest() yielded expected result for " + alg + ":" + size);
+                }, function(err) {
+                    assert_unreached("digest() threw an error for " + alg + ":" + size + " - " + err.message);
+                });
+
+                all_promises.push(promise);
+                return promise;
+            }, mixedCase + " with " + size + " source data");
+
+            promise_test(function(test) {
+                var copiedBuffer = copyBuffer(sourceData[size]);
+                var promise = subtle.digest({name: upCase}, copiedBuffer)
+                .then(function(result) {
+                    copiedBuffer[0] = 255 - copiedBuffer;
+                    assert_true(equalBuffers(result, digestedData[alg][size]), "digest() yielded expected result for " + alg + ":" + size);
+                }, function(err) {
+                    assert_unreached("digest() threw an error for " + alg + ":" + size + " - " + err.message);
+                });
+
+                all_promises.push(promise);
+                return promise;
+            }, upCase + " with " + size + " source data and altered buffer after call");
+
+        });
+    });
+
+    // Call digest() with bad algorithm names to get an error
+    var badNames = ["AES-GCM", "RSA-OAEP", "PBKDF2", "AES-KW"];
+    Object.keys(sourceData).forEach(function(size) {
+        badNames.forEach(function(badName) {
+
+            promise_test(function(test) {
+                var promise = subtle.digest({name: badName}, sourceData[size])
+                .then(function(result) {
+                    assert_unreached("digest() should not have worked for " + alg + ":" + size);
+                }, function(err) {
+                    assert_equals(err.message, "OperationError", "Bad algorithm name should cause OperationError")
+                });
+
+                all_promises.push(promise);
+                return promise;
+            }, badName + " with " + size);
+
+        });
+    });
+
+
+    Promise.all(all_promises).then(function() {done();}, function() {done();});
```

Comment:
This can use the normal done processing rather than the delayed processing.  I don't see any tests that you are setting up inside of a promise.